examples/pytorch/seal/utils.py

```diff
@@ -0,0 +1,161 @@
+    parser = argparse.ArgumentParser(description='SEAL')
+    parser.add_argument('--dataset', type=str, default='ogbl-collab')
+    parser.add_argument('--gpu_id', type=int, default=0)
+    parser.add_argument('--hop', type=int, default=1)
+    parser.add_argument('--model', type=str, default='dgcnn')
+    parser.add_argument('--gcn_type', type=str, default='gcn')
+    parser.add_argument('--num_layers', type=int, default=3)
+    parser.add_argument('--hidden_units', type=int, default=32)
+    parser.add_argument('--sort_k', type=int, default=30)
+    parser.add_argument('--pooling', type=str, default='sum')
+    parser.add_argument('--dropout', type=str, default=0.5)
+    parser.add_argument('--hits_k', type=int, default=50)
+    parser.add_argument('--lr', type=float, default=0.0001)
+    parser.add_argument('--neg_samples', type=int, default=1)
+    parser.add_argument('--subsample_ratio', type=float, default=0.1)
+    parser.add_argument('--epochs', type=int, default=60)
+    parser.add_argument('--batch_size', type=int, default=32)
+    parser.add_argument('--eval_steps', type=int, default=5)
+    parser.add_argument('--num_workers', type=int, default=32)
+    parser.add_argument('--random_seed', type=int, default=2021)
+    parser.add_argument('--save_dir', type=str, default='./processed')
+    args = parser.parse_args()
+
+    return args
+
+
+def load_ogb_dataset(dataset):
+    """
+    Load OGB dataset
+    Args:
+        dataset(str): name of dataset (ogbl-collab, ogbl-ddi, ogbl-citation)
+
+    Returns:
+        graph(DGLGraph): graph
+        split_edge(dict): split edge
+
+    """
+    dataset = DglLinkPropPredDataset(name=dataset)
+    split_edge = dataset.get_edge_split()
+    graph = dataset[0]
+
+    return graph, split_edge
+
+
+def coalesce_graph(graph, aggr_type='sum', copy_data=False):
+    """
+    Coalesce multi-edge graph
+    Args:
+        graph(DGLGraph): graph
+        aggr_type(str): type of aggregator for multi edge weights
+        copy_data(bool): if copy ndata and edata in new graph
+
+    Returns:
+        graph(DGLGraph): graph
+
+
+    """
+    src, dst = graph.edges()
+    graph_df = pd.DataFrame({'src': src, 'dst': dst})
+    graph_df['edge_weight'] = graph.edata['edge_weight'].numpy()
+
+    if aggr_type == 'sum':
+        tmp = graph_df.groupby(['src', 'dst'])['edge_weight'].sum().reset_index()
+    elif aggr_type == 'mean':
+        tmp = graph_df.groupby(['src', 'dst'])['edge_weight'].mean().reset_index()
+    else:
+        raise ValueError("aggr type error")
+
+    if copy_data:
+        graph = dgl.to_simple(graph, copy_ndata=True, copy_edata=True)
+    else:
+        graph = dgl.to_simple(graph)
+
+    src, dst = graph.edges()
+    graph_df = pd.DataFrame({'src': src, 'dst': dst})
+    graph_df = pd.merge(graph_df, tmp, how='left', on=['src', 'dst'])
+    graph.edata['edge_weight'] = torch.from_numpy(graph_df['edge_weight'].values).unsqueeze(1)
+
+    graph.edata.pop('count')
+    return graph
```

Comment:
I suppose we can remove this function now.

Reply:
removed